src/features/foods/components/DeleteFoodDialogContent.tsx

```diff
@@ -60,16 +62,20 @@ export const DeleteFoodDialogContent = ({
    * @returns void
    */
   const handleDelete = async () => {
+    if (isLoading) return;
+    setIsLoading(true);
     if (!containerId || !foodId) return;
     const result = await removeFood(containerId, foodId);
     if (!result.ok) {
       alert('Something went wrong. Please try again.');
       onDialogClose();
       onParentClose?.();
+      setIsLoading(false);
     } else {
       alert('Successfully deleted!');
       onDialogClose();
       onParentClose?.();
+      setIsLoading(false);
       router.refresh();
     }
   };
```

Comment:
not only in this file, but is there any specifc reason `setIsLoading(false)` is in not only inside `if` statemene, but also `else` statement? I thought it is not depend on `if-else`,so it can be written like below??

```tsx
 if (!result.ok) {
      alert('Something went wrong. Please try again.');
      onDialogClose();
      onParentClose?.();
    } else {
      alert('Successfully deleted!');
      onDialogClose();
      onParentClose?.();
      router.refresh();
    }
    setIsLoading(false);
```

Reply:
The reason why `setIsLoading(false)` is placed within the `if-else` statement is because `router.refresh()` should be called at the very end of the event handler. If you take `setIsLoading(false)` out the `if-else`, you will need another `if-else` statement in the case.

Reply:
@nick-y-ito Thanks for explanation, I'm clear
Other than that, LGTM! thanks for your work!!